commcare_connect/opportunity/models.py

```diff
@@ -276,6 +277,30 @@ def display_name(self):
         else:
             return "---"
 
+    @property
```

Comment:
there is a `cached_property` decorator in django that should handle this https://docs.djangoproject.com/en/5.0/ref/utils/#module-django.utils.functional

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/302/commits/8511d645903cf17689677fda64d2e561a18ff197